core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/authorization/PermissionTree.java

```diff
@@ -169,6 +178,9 @@ void denyRoleInTree(String roleName, String resourceId, String action, boolean u
             if (updateCache) {
                 invalidateCache(root);
             }
+        } catch (Exception e) {
```

Comment:
The db.commit() throws SQLException 
authorizeRoleInTree method and denyRoleInTree method throws RuntimeException.
We can have the two exception handling seperately. But dbconnection should be rolled back when and exception is thrown. That is why it catches Exception.